tool/tsh/tsh.go

```diff
@@ -4267,13 +4271,75 @@ func forEachProfile(cf *CLIConf, fn func(tc *client.TeleportClient, profile *cli
 	return trace.NewAggregate(errors...)
 }
 
+type updateKubeConfigOpt struct {
+	warnOnDeprecatedSNI bool
+	warnSNIPrinted      *bool
```

Comment:
```suggestion
        // warnSNIPrinted is a pointer so that it defaults to nil, which is treated as true
	warnSNIPrinted      *bool
```

Reply:
this is still confusing, maybe change to `disableSNIWarning` if you want to warn by default?

Reply:
I have added a comment explaning the usage of `warnSNIPrinted`